substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/ObjectFile.java

```diff
@@ -1289,7 +1297,7 @@ public Element getOffsetBootstrapElement() {
 
     private final HashSet<LayoutDecision> allDecisions = new HashSet<>();
     private final Map<Element, LayoutDecisionMap> decisionsByElement = new IdentityHashMap<>();
-    private final Map<Element, LayoutDecisionMap> decisionsTaken = new IdentityHashMap<>();
+    public final Map<Element, LayoutDecisionMap> decisionsTaken = new IdentityHashMap<>();
```

Comment:
This field appears only to be needed by `ObjectFile` and subclass `MachOObjectFile`. So, it would be sensible to make it `protected` rather than `public`. Even better might be to offer a protected lookup method:

    protected Object getLayoutDecisionValue(Element e, LayoutDecision.Kind k) {
        return decisionsTaken.get(e).getDecidedValue(k));
    }